File: archive-application/src/main/java/site/archive/dto/v2/ArchiveCommunityResponseDto.java
package site.archive.dto.v2;

import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import site.archive.common.DateTimeUtils;
import site.archive.domain.archive.Archive;
import site.archive.domain.archive.CoverImageType;
import site.archive.domain.archive.Emotion;

@RequiredArgsConstructor
@Builder
@Getter
public class ArchiveCommunityResponseDto {

    private final Long archiveId;
    private final String name;
    private final String watchedOn;
    private final Emotion emotion;
    private final String mainImage;
    private final CoverImageType coverImageType;
    private final Long authorId;
    private final String authorNickname;
    private final String authorProfileImage;
    private final Boolean isLiked;
    private final Long likeCount;
    private final Long dateMilli;

    public static ArchiveCommunityResponseDto from(Archive archive, Long currentUserIdx, long dateMilli) {
        var author = archive.getAuthor();
        var isLiked = archive.getLikes().stream()
                             .anyMatch(like -> !like.getIsDeleted() && like.getUser().getId().equals(currentUserIdx));
        var likeCount = archive.getLikes().stream()
                               .filter(like -> !like.getIsDeleted()).count();
        return ArchiveCommunityResponseDto.builder()
                                          .archiveId(archive.getId())
                                          .name(archive.getName())
                                          .watchedOn(archive.getWatchedOn().format(DateTimeUtils.getYymmddFormatter()))
                                          .emotion(archive.getEmotion())
                                          .mainImage(archive.getMainImage())
                                          .coverImageType(archive.getCoverImageType())
                                          .authorId(author.getId())
                                          .authorNickname(author.getNickname())
                                          .authorProfileImage(author.getProfileImage())
                                          .isLiked(isLiked)
                                          .likeCount(likeCount)
                                          .dateMilli(dateMilli)
                                          .build();
    }

}

File: archive-application/src/main/kotlin/site/archive/dto/v2/ArchiveCommunityResponseDto.kt
package site.archive.dto.v2

import site.archive.common.yymmddFormatter
import site.archive.domain.archive.Archive
import site.archive.domain.archive.CoverImageType
import site.archive.domain.archive.Emotion

class ArchiveCommunityResponseDto(
    val archiveId: Long,
    val name: String,
    val watchedOn: String,
    val emotion: Emotion,
    val mainImage: String,
    val coverImageType: CoverImageType,
    val authorId: Long,
    val authorNickname: String,
    val authorProfileImage: String,
    val isLiked: Boolean,
    val likeCount: Long,
    val dateMilli: Long
) {
    companion object {
        @JvmStatic
        fun from(archive: Archive, currentUserIdx: Long, dateMilli: Long): ArchiveCommunityResponseDto {
            val author = archive.author
            val isLiked = archive.likes.stream()
                .anyMatch { !it.isDeleted && it.user.id.equals(currentUserIdx) }
            val likeCount = archive.likes.stream()
                .filter { !it.isDeleted }
                .count()
            return ArchiveCommunityResponseDto(
                archiveId = archive.id,
                name = archive.name,
                watchedOn = archive.watchedOn.format(yymmddFormatter),
                emotion = archive.emotion,
                mainImage = archive.mainImage,
                coverImageType = archive.coverImageType,
                authorId = author.id,
                authorNickname = author.nickname,
                authorProfileImage = author.profileImage,
                isLiked = isLiked,
                likeCount = likeCount,
                dateMilli = dateMilli
            )
        }
    }
}